### DISCUSSION TITLE
brew upgrade did not work on macOS 14 beta (Sonoma)

### DISCUSSION THREAD

**Discussion by elisezhu123** [Homebrew/discussions#4614] [Getting started]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.26
ORIGIN: https://github.com/Homebrew/brew
HEAD: eff45ef570f265e226f14ce91da72d7a6e7d516a
Last commit: 4 days ago
Core tap JSON: 01 Jul 17:20 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.41.0 => /opt/homebrew/bin/git
Curl: 8.0.1 => /usr/bin/curl
macOS: 14.0-arm64
CLT: 15.0.0.0.1.1686949477
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You are using macOS 14.
We do not provide support for this pre-release version.
It is expected behaviour that some formulae will fail to build in this pre-release version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

it just shows 
You are using macOS 14.
We do not provide support for this pre-release version.
It is expected behaviour that some formulae will fail to build in this pre-release version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.

**Comment by Bo98** [comment]
This is a warning that will always display, but it won't fail anything on its own. If it's failing, it's for another reason than that message.

**Comment by pawel-krupinski-bla** [comment]
I upgraded from beta to release Mac OSx 14 and I'm still getting this warning.

**Reply by SMillerDev** [reply]
MacOS Sonoma won't be released until the 26th so that's not possible.

**Comment by Moranilt** [comment]
I have the same error on MacOS 14 (not beta)

**Reply by SMillerDev** [reply]
What does your `brew config` say?

**Reply by may3rr** [reply]
have you fix it? @_@

**Comment by interglobalmedia** [comment]
Hi,

I recently upgraded to macOS Sonoma, and I am now having problems updating and upgrading Homebrew. When I tried to run brew update, I got back the following:

`git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Error: Fetching /opt/homebrew failed!
Fetching /opt/homebrew/Library/Taps/mongodb/homebrew-brew failed!
Fetching /opt/homebrew/Library/Taps/ngrok/homebrew-ngrok failed!
Fetching /opt/homebrew/Library/Taps/railwaycat/homebrew-emacsmacport failed!
Fetching /opt/homebrew/Library/Taps/shivammathur/homebrew-php failed!`

And when I tried to upgrade Homebrew, it got stuck at one point and did not complete the process. Please advise. Thanks!

**Reply by SMillerDev** [reply]
What does `brew doctor` and `brew config` say?

**Reply by interglobalmedia** [reply]
Hi,

I actually found my answer on stackoverflow. I just had to comment out the following line(s) in my `.gitconfig`:

`#   insteadOf = https://github.com/`

And now it works!

**Reply by okineadev** [reply]
> Hi,
> 
> I actually found my answer on stackoverflow. I just had to comment out the following line(s) in my `.gitconfig`:
> 
> `#   insteadOf = https://github.com/`
> 
> And now it works!

Please provide a link to this solution

**Comment by Amponsem** [comment]
Run these commands
```sh
brew tap --repair
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

**Reply by SkywalkerH2002** [reply]
This works perfectly for me. Thanks!
